docs/api.md

```diff
@@ -5,31 +5,88 @@ permalink: /docs/api/
 ---
 
 ##Compiling a template
+###Function call
+```    
+dust.compile(content, templateName);
+```
+###Parameters
+* content {String}: The content of your dust template.
+* templateName {String}: The name under which the template will be registered.  This is the name you will use when rendering the template.
+* @return {String} compiled javascript output of the template.
+
+###Example call
 Assuming you have the dust compiler in your JS environment, you can compile a Dust template source file to JavaScript form. The following will compile this very simple template "Hello {name}!" to a JavaScript string that will also register it as a template named "intro".
 
 ```
 var compiledTemplate = dust.compile("Hello {name}!", "intro");
 ```
 
-If you include the "compiled" string as part of a script block of JS that you load, then the "intro" template will be defined and registered. If you want to do it immediately then do:
+If you include the "compiled" output as part of a script block of JS that you load, then the "intro" template will be defined and registered. 
 
 ```
-dust.loadSource(compiled);
+<script type="text/javascript">
+/* example compiled dust template output */
+(function() {
+  dust.register("demo", body_0);
+
+  function body_0(chk, ctx) {
+    return chk.write("Hello ").reference(ctx.get("name"), ctx, "h").write("! You have ").reference(ctx.get("count"), ctx, "h").write(" new messages.");
+  }
+  return body_0;
+})();
+</script>
+```
+
+If you want to register it directly from the ouput of the dust.compile function, use dust.loadSource:
+
+```
+var compiledTemplate = dust.compile("Hello {name}!", "intro");
+dust.loadSource(compiledTemplate);
 ```
 
 This will also define and register the template.  All of these compiled templates are saved in the dust.cache Array under the name they were registered under, in this case, "intro".
 
 ##Rendering a template
-Assuming you have the "intro" template we previously compiled and loaded, you can render it using the code below. The first argument to dust.render is the registered template name, the second argument is the JSON model, the third argument is a callback function where "out" contains the rendered output from running the template (and err has any error information).
+###Function call
+```    
+dust.render(templateName, data, callback);
+```
+###Parameters
+* templateName {String}: The registered name of the template to render.  This was the name given to the template at the compilation step.
+* data {Object}: The javascript context to be consumed by the template.
```

Comment:
What does "consumed" me. How about "The data to be used to populate the template"?